### PR TITLE
LanguageModuleRequirement: remove rbx support

### PR DESCRIPTION
**Pull request by mistydemeo**
This test wasn't running by default, so we missed that it wasn't actually being executed - or that it was failing when running in the testing environment.

As far as I can tell this is not, and has not, been used either in core or in any tap, third party or otherwise, so just remove the feature and its test.

### CODE REVIEW

**Comment by DomT4** [comment]
:+1:. Makes sense to me, and means I can drop the additional check from #42576 which I've recently started working on again.

**Comment by mistydemeo** [comment]
Updated with commits to also remove chicken, node, ocaml, and jruby support. I was unable to find evidence any of these are in use, and only scare evidence they were ever used. Chicken and node also had iffy tests, and ocaml and jruby had no tests.

File: Library/Homebrew/dependency_collector.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
 class DependencyCollector
   # Define the languages that we can handle as external dependencies.
   LANGUAGE_MODULES = Set[
-    :chicken, :jruby, :lua, :node, :ocaml, :perl, :python, :python3, :rbx, :ruby
+    :lua, :perl, :python, :python3, :ruby
```
Comment (DomT4) [inline Library/Homebrew/dependency_collector.rb]: Should `:lua51` be part of this list? Did I forget to add it at the time?

Comment (mistydemeo) [inline Library/Homebrew/dependency_collector.rb]: Looks like it was omitted, yeah. I see no evidence it's been used (though `:lua` is).

Comment (DomT4) [inline Library/Homebrew/dependency_collector.rb]: Garg. Bad Dom.

Yeah, it's a relatively recent addition. There was a big Issue 6 or so months ago with "Homebrew supports Lua terribly" which was a fair comment, so I've been extremely slowly working on improving that (Though not a priority of mine, tbh).

The addition here was essentially a workaround for people who wanted it whilst investigated the possibility of whether we could do something similar to how we handle Python these days.

Comment (mistydemeo) [inline Library/Homebrew/dependency_collector.rb]: OK, added a commit which adds it to the set.

Comment (DomT4) [inline Library/Homebrew/dependency_collector.rb]: Thanks Misty. I'll try and stick it back on my pile at some point soon so it doesn't sit there indefinitely without longer-term justification.

**Comment by xu-cheng** [comment]
Sorry, late for comments. But I wonder should we relocate these codes to `compat` folder instead of just deleting them.

**Pull request closed without merging**